src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,250 @@
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+            else:
+                logging.info("HoldTime conformance failed.  Test step skipped")
+                asserts.fail("HoldTime conformance is incorrect")
+
+        else:
+            logging.info("HoldTimeLimits not supported. Test step skipped")
+            self.mark_current_step_skipped()
+        
+        self.step(6)
+        if attributes.HoldTime.attribute_id in attribute_list:
+            hold_time_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+            hold_time_limits_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTimeLimits)
+
+            asserts.assert_less_equal(hold_time_dut, hold_time_limits_dut.HoldTimeMax, "HoldTime attribute is out of range")
+            asserts.assert_greater_equal(hold_time_dut, hold_time_limits_dut.HoldTimeMin, "HoldTime attribute is out of range")
+        else:
+            logging.info("HoldTime not supported. The rest of legacy attribute test can be skipped")
+            self.skip_all_remaining_steps(7)
+            return
+
+        self.step(7)
+        if attributes.PIROccupiedToUnoccupiedDelay.attribute_id in attribute_list:
+            if ((occupancy_sensor_type_bitmap_dut == 0b00000001) | ((occupancy_sensor_type_bitmap_dut != 0b00000001)&(occupancy_sensor_type_bitmap_dut != 0b00000010)&(occupancy_sensor_type_bitmap_dut != 0b00000100))):
+                pir_otou_delay_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.PIROccupiedToUnoccupiedDelay)
+
+                asserts.assert_less_equal(pir_otou_delay_dut, 0xFFFE, "PIROccupiedToUnoccupiedDelay is not in valid range")
+                asserts.assert_greater_equal(pir_otou_delay_dut, 0, "PIROccupiedToUnoccupiedDelay is not in valid range")
+            else:
+                logging.info("PIROccupiedToUnoccupiedDelay conformance failed")
+                asserts.fail("PIROccupiedToUnoccupiedDelay conformance is incorrect")
+        else:
+            logging.info("PIROccupiedToUnoccupiedDelay not supported. Test step skipped")
+            self.mark_current_step_skipped()
+
+        self.step(8)
+        if attributes.PIRUnoccupiedToOccupiedDelay.attribute_id in attribute_list: 
+            has_delay = attributes.PIRUnoccupiedToOccupiedDelay.attribute_id in attribute_list
+            has_threshold = attributes.PIRUnoccupiedToOccupiedThreshold.attribute_id in attribute_list
+            asserts.assert_equal(has_delay, has_threshold, "PIRUnoccupiedToOccupiedDelay conformance failure")

```

Comment:
this is not in the test plan (and applies throughout). Test plan copy I have is:

![image](https://github.com/user-attachments/assets/0c081c80-75da-4e9c-a52e-84b5f68fc9af)

So we are only to read the value and check ranges. Do we have to check delay/threshold? If so we should comment why we do this on top of the test plan.

Reply:
This test case just check the presence of the attribute and if it is within the date type that is defined.  The content of the attribute will not be tested in this test plan.  This attribute is old Zigbee legacy timing attribute and will be deprecated in the future version.  But we do check it with HoldTime for the support of backward compatibility.  The actual timing functionality is tested on HoldTime.